modules/fedimint-mint-client/src/backup/recovery.rs

```diff
@@ -411,21 +426,19 @@ impl MintRecoveryState {
         // greedy no matter what and take what we can, and just report
         // anything suspicious.
 
-        if let Some((issuance_request, note_idx, pending_amount)) = self
-            .pending_nonces
-            .get(&output.blind_nonce.0.into())
-            .copied()
+        if let Some((issuance_request, note_idx, pending_amount)) =
+            self.pending_nonces.remove(&output.blind_nonce.0.into())
         {
             // the moment we see our blind nonce in the epoch history, correctly or
             // incorrectly used, we know that we must have used
             // already
             self.observe_nonce_idx_being_used(pending_amount, note_idx, secret);
 
             if pending_amount == output.amount {
-                assert!(self
-                    .pending_nonces
-                    .remove(&output.blind_nonce.0.into())
-                    .is_some());
```

Comment:
Moved into the `if` head. Valid transformation because we re-add in the else branch (which was a no-op previously).

Reply:
Not true anymore after merging #6096

Reply:
Yes. Added it back.